homeassistant/components/sensor/snmp.py

```diff
@@ -0,0 +1,134 @@
+"""
+Support for displaying collected data over SNMP.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.snmp/
+"""
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (CONF_HOST, CONF_PLATFORM, CONF_NAME,
+                                 CONF_PORT)
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pysnmp==4.3.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "SNMP"
+DEFAULT_COMMUNITY = "public"
+DEFAULT_PORT = "161"
+CONF_COMMUNITY = "community"
+CONF_BASEOID = "baseoid"
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'snmp',
+    vol.Optional(CONF_NAME): vol.Coerce(str),
+    vol.Required(CONF_HOST): vol.Coerce(str),
+    vol.Optional(CONF_PORT): vol.Coerce(int),
+    vol.Optional(CONF_COMMUNITY): vol.Coerce(str),
+    vol.Required(CONF_BASEOID): vol.Coerce(str),
+    vol.Optional('unit_of_measurement'): vol.Coerce(str),
```

Comment:
> Why can the user pick a unit of measurement ?

Because certain nodes inside the tree are delivered without unit of measurement. Others are. Anyway, in a lot of cases it will be necessary to use a template sensor for the conversion in a more human readable form as long that the template support for this sensor is missing. 

> Also, we have a constant for that in `homeassistant.const`.

Thanks, was only checking CONF* and not ATTR*.